internal/pkg/cli/task_run.go

```diff
@@ -0,0 +1,242 @@
+
+	// sets up Dockerfile parser using fs and input path
+	setupParser func(opts *runTaskOpts)
+}
+
+func newTaskRunOpts(vars runTaskVars) (*runTaskOpts, error) {
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("new config store: %w", err)
+	}
+
+	return &runTaskOpts{
+		runTaskVars: vars,
+
+		fs:    &afero.Afero{Fs: afero.NewOsFs()},
+		store: store,
+
+		setupParser: func(o *runTaskOpts) {
+			o.df = dockerfile.New(o.fs, o.DockerfilePath)
+		},
+	}, nil
+}
+
+// Validate returns an error if the flag values passed by the user are invalid.
+func (o *runTaskOpts) Validate() error {
+	if o.Num == 0 {
+		return errors.New("number of tasks must be positive")
+	}
+
+	if o.CPU == 0 {
+		return errors.New("CPU units must be positive")
+	}
+
+	if o.Memory == 0 {
+		return errors.New("memory must be positive")
+	}
+
+	if o.Image != "" && o.DockerfilePath != "" {
+		return errors.New("cannot specify both image and dockerfile path")
+	}
+
+	if o.Image != "" {
+		if err := o.validateImageName(); err != nil {
+			return err
+		}
+	}
+
+	if o.DockerfilePath != "" {
+		if _, err := o.fs.Stat(o.DockerfilePath); err != nil {
+			return err
+		}
+	}
+
+	if o.Env != "" && (o.SubnetID != "" || o.SecurityGroupIDs != nil) {
+		return errors.New("can only specify one of a)env and b)subnet id and (or) security groups")
+	}
+
+	if o.SubnetID != "" {
+		if err := o.validateSubnetID(); err != nil {
+			return err
+		}
+	}
+
+	if o.SecurityGroupIDs != nil {
+		if err := o.validateSecurityGroupIDs(); err != nil {
+			return err
+		}
+	}
+
+	if o.App != "" {
+		if err := o.validateAppName(); err != nil {
+			return err
+		}
+	}
+
+	if o.Env != "" {
+		if err := o.validateEnvName(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *runTaskOpts) validateImageName() error {
+	valid, err := regexp.MatchString(`^\d+\.dkr\.ecr\.[a-z0-9\-]+.amazonaws.com/[a-z][a-z0-9\-]*$`, o.Image)
+	if err != nil {
+		return fmt.Errorf("validate image name: %w", err)
+	}
+
+	if !valid {
+		return errors.New("image name is malformed")
+	}
+
+	return nil
+}
+
+func (o *runTaskOpts) validateSubnetID() error {
+	valid, err := regexp.MatchString(`^subnet-[a-z0-9]+`, o.SubnetID)
+	if err != nil {
+		return fmt.Errorf("validate subnet id: %w", err)
+	}
+
+	if !valid {
+		return errors.New("subnet id is malformed")
+	}
+
+	return nil
+}
+
+func (o *runTaskOpts) validateSecurityGroupIDs() error {
+	for _, id := range o.SecurityGroupIDs {
+		valid, err := regexp.MatchString(`^sg-[a-z0-9]+`, id)
+		if err != nil {
+			return fmt.Errorf("validate security group ids: %w", err)
+		}
+
+		if !valid {
+			return errors.New("one or more malformed security group id(s)")
+		}
+	}
+	return nil
+}
+
+func (o *runTaskOpts) validateAppName() error {
+	if o.AppName() == "" {
+		if _, err := o.store.GetApplication(o.App); err != nil {
+			return fmt.Errorf("get application: %w", err)
+		}
+	}
+	return nil
+}
+
+func (o *runTaskOpts) validateEnvName() error {
```

Comment:
This validation, on the other hand, is good to leave in the task_run.go file as a struct method since it relies on the store type to get available environments.